### ISSUE TITLE
Syntax Highlighting for module names and constructors

### ISSUE THREAD

**Issue by Chadtech** [klazuka/intellij-elm#439]
Hi,

In other editors, I have been able to set my Elm syntax highlights up, such that module names and type constructors get their own coloring. I think this really helps because constructors and module names are different a different kind of thing than just regular value and function names.

I am proposing:
- Module names used in function bodies get their own coloring (like `Model` in `Model.mapTracker` in my picture below)
- type constructors get their own coloring (like `BeatMsg` in the function body, in the case statement, and in the type definition of `Msg`, in my picture below)
![DANK MEME 2019-07-20 at 18 31 50](https://user-images.githubusercontent.com/2939686/61581293-acea9700-ab1c-11e9-891c-eb2145053c47.png)

This picture comes from my vs code editor, which I have manager to add this syntax highlighting for.

For reference, heres the same code snippet, with the same colors and fonts, without this syntax highlighting, in intellij :
![DANK MEME 2019-07-20 at 18 33 12](https://user-images.githubusercontent.com/2939686/61581305-dd323580-ab1c-11e9-9265-f2e9ac94a00f.png)

What do you think? I'm not sure how the syntax highlighting code works in this plugin, but if you think my proposal is a good idea, I would be happy to try and make the PR myself.

**Comment by klazuka** [comment]
Yes, this would definitely be an improvement. This is a dupe of #25, but this one has more detail so I will keep it open. I had intended to implement this way back in the early days, but got side-tracked by the mountain of other tasks.

If you want to implement it, the code would go in `org.elm.ide.highlight.ElmSyntaxHighlightAnnotator`. And you would want to register new "colors" for modules and union constructors in `org.elm.ide.color.ElmColor`. And then the last thing is wiring it up to the color settings UI by modifiying the demo text and the color mapping in `org.elm.ide.color.ElmColorSettingsPage#getAdditionalHighlightingTagToDescriptorMap`

**Comment by klazuka** [comment]
See comment on #450 regarding the difficulties I had styling the module qualifier.
